Update to newer Vertica DB for tests

Right now we're testing against Vertica 7.1.2, but 9.2.1 is out. 

We'll need to use a newer Docker image for CI. The one we're currently using seems like it's no longer being actively developed/ [jbfavre/vertica:latest](https://github.com/jbfavre/docker-vertica) looks like a reasonable replacement, I was able to drop it in without having to change any env vars. We will also need to switch the JDBC driver version used in CI.

I switched to a newer 9.x JDBC driver, but I've ran into a few problems when running tests: 

```
Caused by: com.vertica.support.exceptions.DataException: [Vertica][VJDBC](3787) ROLLBACK: Invalid value for parameter
 at com.vertica.util.ServerErrorData.buildException (:-1)
    com.vertica.dataengine.VResultSet.fetchChunk (:-1)
    com.vertica.dataengine.VResultSet.initialize (:-1)
    com.vertica.dataengine.VQueryExecutor.readExecuteResponse (:-1)
    com.vertica.dataengine.VQueryExecutor.handleExecuteResponse (:-1)
    com.vertica.dataengine.VQueryExecutor.execute (:-1)
    com.vertica.jdbc.common.SPreparedStatement.executeWithParams (:-1)
    com.vertica.jdbc.common.SPreparedStatement.executeUpdate (:-1)
```

This was happening when loading the test data. So we'll have to work out why that's happening in order to upgrade.

--- a/.github/workflows/drivers.yml
+++ b/.github/workflows/drivers.yml
@@ -663,15 +663,13 @@ jobs:
       DRIVERS: vertica
     services:
       vertica:
-        image: sumitchawla/vertica
+        image: vertica/vertica-ce:12.0.2-0
         ports:
           - "5433:5433"
     steps:
     - uses: actions/checkout@v3
     - name: Make plugins directory
       run: mkdir plugins
-    - name: Fetch JDBC driver
-      run: wget --output-document=plugins/vertica-jdbc-7.1.2-0.jar ${{ secrets.VERTICA_JDBC_JAR }}
     - name: Test Vertica driver
       uses: ./.github/actions/test-driver
       with:

--- a/modules/drivers/vertica/deps.edn
+++ b/modules/drivers/vertica/deps.edn
@@ -2,7 +2,7 @@
  ["src" "resources-ee"]
 
  :deps
- {com.vertica.jdbc/vertica-jdbc {:mvn/version "10.0.1-0"}}
+ {com.vertica.jdbc/vertica-jdbc {:mvn/version "12.0.3-0"}}
 
  :aliases
  {:oss

--- a/modules/drivers/vertica/test/metabase/test/data/vertica.clj
+++ b/modules/drivers/vertica/test/metabase/test/data/vertica.clj
@@ -38,7 +38,7 @@
   (defmethod sql.tx/field-base-type->sql-type [:vertica base-type] [_ _] sql-type))
 
 (defn- db-name []
-  (tx/db-test-env-var-or-throw :vertica :db "docker"))
+  (tx/db-test-env-var-or-throw :vertica :db "VMart"))
 
 (def ^:private db-connection-details
   (delay {:host     (tx/db-test-env-var-or-throw :vertica :host "localhost")

--- a/test/metabase/query_processor_test/share_test.clj
+++ b/test/metabase/query_processor_test/share_test.clj
@@ -32,14 +32,22 @@
                                    [:ends-with $name "t"]]]]]})))))
 
     (testing "empty results"
-      ;; due to a bug in the Mongo counts are returned as empty when there are no results (#5419)
-      (is (= (if (= driver/*driver* :mongo)
-               []
-               [[nil]])
-             (mt/rows
-               (mt/run-mbql-query venues
-                 {:aggregation [[:share [:< $price 4]]]
-                  :filter      [:> $price Long/MAX_VALUE]})))))))
+      ;; Vertica doesn't allow dividing null by zero
+      ;; TODO consider wrapping all divisions in nullif checking the first argument
+      (if (= driver/*driver* :vertica)
+        (is (thrown-with-msg?
+             clojure.lang.ExceptionInfo #"Division by zero"
+             (mt/run-mbql-query venues
+               {:aggregation [[:share [:< $price 4]]]
+               :filter      [:> $price Long/MAX_VALUE]})))
+        ;; due to a bug in the Mongo counts are returned as empty when there are no results (#5419)
+        (is (= (if (= driver/*driver* :mongo)
+                 []
+                 [[nil]])
+               (mt/rows
+                (mt/run-mbql-query venues
+                  {:aggregation [[:share [:< $price 4]]]
+                   :filter      [:> $price Long/MAX_VALUE]}))))))))
 
 (deftest segments-metrics-test
   (mt/test-drivers (mt/normal-drivers-with-feature :basic-aggregations)

